### PR TITLE
Further fixes for 5.0.0 compatibility

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

There is still one issue left which I am working through, but this PR:

- Allows the `neo4j-driver@5.0.0` to be used as a peer dependency
- Generates `CREATE CONSTRAINT` calls dynamically based on database version
- Removes some use of index functions from tests

### CODE REVIEW

File: packages/graphql/tests/integration/directives/fulltext.int.test.ts [hunk 1]
```diff
@@ -570,7 +564,7 @@ describe("@fulltext directive", () => {
         const session = driver.session({ database: databaseName });
 
         const cypher = `
-            CALL db.indexes() yield
+            SHOW INDEXES yield
```
Comment (MacondoExpress) [inline packages/graphql/tests/integration/directives/fulltext.int.test.ts:567]: Do we still have to support the 4.1 version?
I think that `SHOW INDEXES` is available only from the 4.2 
https://neo4j.com/docs/cypher-manual/4.2/deprecations-additions-removals-compatibility/

Comment (darrellwarde) [reply packages/graphql/tests/integration/directives/fulltext.int.test.ts:567]: Nope, only 4.3 onwards. 🙂

Comment (MacondoExpress) [reply packages/graphql/tests/integration/directives/fulltext.int.test.ts:567]: Cool!

**Review by MacondoExpress — APPROVED** [review]
LGTM!!

**Pull request merged**